From 33391f8b0a2ccda3a27d95dfe67846d5c60666f7 Mon Sep 17 00:00:00 2001
From: Alan Tull <r80115@freescale.com>
Date: Fri, 2 Jul 2010 08:29:19 -0500
Subject: [PATCH] ENGR00124852 sgtl5000: support nonstandard sample rates

Supporting more sample rates that are not listed in pcm.h
(12KHz and 24KHz).

Signed-off-by: Alan Tull <r80115@freescale.com>
---
 sound/soc/codecs/sgtl5000.c |    7 ++++++-
 sound/soc/imx/imx-ssi.c     |    2 +-
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/sound/soc/codecs/sgtl5000.c b/sound/soc/codecs/sgtl5000.c
index 1016c4b..a780670 100644
--- a/sound/soc/codecs/sgtl5000.c
+++ b/sound/soc/codecs/sgtl5000.c
@@ -596,6 +596,10 @@ static int sgtl5000_pcm_hw_params(struct snd_pcm_substream *substream,
 	case 22050:
 		sys_fs = 44100;
 		break;
+	case 12000:
+	case 24000:
+		sys_fs = 48000;
+		break;
 	default:
 		sys_fs = sgtl5000->lrclk;
 		break;
@@ -846,7 +850,8 @@ static int sgtl5000_set_bias_level(struct snd_soc_codec *codec,
 		      SNDRV_PCM_RATE_32000 |\
 		      SNDRV_PCM_RATE_44100 |\
 		      SNDRV_PCM_RATE_48000 |\
-		      SNDRV_PCM_RATE_96000)
+		      SNDRV_PCM_RATE_96000 |\
+		      SNDRV_PCM_RATE_KNOT)
 
 #define SGTL5000_FORMATS (SNDRV_PCM_FMTBIT_S16_LE |\
 			SNDRV_PCM_FMTBIT_S20_3LE |\
diff --git a/sound/soc/imx/imx-ssi.c b/sound/soc/imx/imx-ssi.c
index 027e32d..0ab30fd 100644
--- a/sound/soc/imx/imx-ssi.c
+++ b/sound/soc/imx/imx-ssi.c
@@ -627,7 +627,7 @@ static void imx_ssi_remove(struct platform_device *pdev,
 	SNDRV_PCM_RATE_16000 | SNDRV_PCM_RATE_22050 | \
 	SNDRV_PCM_RATE_32000 | SNDRV_PCM_RATE_44100 | \
 	SNDRV_PCM_RATE_48000 | SNDRV_PCM_RATE_88200 | \
-	SNDRV_PCM_RATE_96000)
+	SNDRV_PCM_RATE_96000 | SNDRV_PCM_RATE_KNOT)
 
 #define IMX_SSI_FORMATS \
 	(SNDRV_PCM_FMTBIT_S16_LE | SNDRV_PCM_FMTBIT_S20_3LE | \
-- 
1.5.4.4

